plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/helpers.ts

```diff
@@ -73,18 +76,20 @@ export const getAndWriteAttachments = async (
   const productArr: string[][] = [];
   const baseUrl = config.getString('confluence.baseUrl');
   const token = config.getString('confluence.token');
+  const isCloud = config.getOptionalBoolean('confluence.isCloud') || false;
+  const authToken = isCloud ? `Basic ${token}` : `Bearer ${token}`;
```

Comment:
For this line, are you storing a base64 encoded username and password as your token variable. I wonder if instead we can update this to take in a username and password and encode it ? That way the user does not have to create the token locally, but we can create it for them? Then we can update the main functionality to do a check that either a token or username/password combo was supplied and if not throw an error.
Access to the non cloud rest api also supports basic auth and is something I was considering adding here soon. Which is why I am suggesting this.  Let me know what you think about this!

Reply:
Yes, I took the path that Roadie.io used for their [Jira plugin](https://roadie.io/backstage/plugins/jira/#get-and-provide-jira_token-as-env-variable) where the user needs to do the encoding to avoid having more configuration (in our case we can now use the same token). I've got instructions on doing this in the README changes. For Confluence Cloud it's not a user name and password but your the email for your Atlassian account and the PAT like this: `<your-atlassian-account-mail>:<your-jira-token>`.

For this PR I'd like to stick with this and then maybe we can follow up with a PR focused on just this part?

Reply:
Sure keep it like this for now. But it will need to be followed up on if there is a desire to allow non cloud users to log in with basic(logically.  A user could in theory set the isCloud to true even if it is not to use basic with an already encoded token!).  But if the precedent is that users do the encoding beforehand, and it is not done in the logic, we can keep the code using just a token and require encoding is done locally, like how you have it set up now.

Reply:
I ended up doing a bit refactor and I've included this now, thanks 🚀